### PR TITLE
br: save global checkpoint to storage and support complete transaction

### PR DESCRIPTION
**Pull request by joccau**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/34632

Problem Summary:

### What is changed and how it works?
- Save global-checkpoint-ts into storage when stop task.
- support check metadata of log dir. it can query `log-min-ts` and `log-max-ts` currently.
- restore the whole complete transaction at start-ts or truncate-ts.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/task/stream.go [hunk 1]
```diff
@@ -836,7 +886,7 @@ func RunStreamTruncate(c context.Context, g glue.Glue, cmdName string, cfg *Stre
 		return nil
 	}
 
-	removed := metas.RemoveDataBefore(cfg.Until)
+	removed := metas.RemoveDataBefore(shiftUntilTS)
```
Comment (3pointer) [inline br/pkg/task/stream.go:889]: is this `shiftUntilTS` distinguish the default cf and write cf file?

Comment (3pointer) [reply br/pkg/task/stream.go:889]: ok, we have distinguish the default cf file in restoration.

**Review by 3pointer — APPROVED** [review]
LGTM

**Review by YuJuncen — APPROVED** [review]

**Pull request merged**